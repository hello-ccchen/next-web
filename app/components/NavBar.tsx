import Link from "next/link";
import Image from "next/image";
import { getServerSession } from "next-auth";
import { authOptions } from "../api/auth/[...nextauth]/options";

const NavBar = async () => {
  const session = await getServerSession(authOptions);
  return (
    <nav className="navbar fixed-top navbar-light bg-light shadow border-bottom border-light">
      <div className="container-fluid">
        <Link className="navbar-brand" href="/">
          <Image src="/next.svg" alt="Home Logo" width={100} height={24} />
        </Link>
        {session && (
          <Link
            className="d-flex justify-content-center py-3 btn"
            href="/api/auth/signout?callbackUrl=/"
          >
            Sign out
            <Image
              src="/logout.svg"
              alt="logout"
              width={24}
              height={24}
              className="ms-2"
            />
          </Link>
        )}
      </div>
    </nav>
  );
};

export default NavBar;
