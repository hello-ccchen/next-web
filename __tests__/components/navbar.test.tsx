/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import { useSession } from "next-auth/react";
import { Session } from "next-auth";
import NavBar from "@/components/navbar";

jest.mock("next-auth/react", () => ({
  useSession: jest.fn(),
}));

describe("NavBar component", () => {
  it("renders `Sign in` button when user's session not exists", () => {
    (useSession as jest.Mock).mockReturnValue({
      status: "unauthenticated",
      data: null,
    });

    const { container, getByText } = render(<NavBar />);
    expect(container).toMatchSnapshot();
    expect(getByText('Sign in')).toBeTruthy();
  });

  it("renders logged-in user's info when user's session exists", () => {
    const mockSession: Session = {
      user: { name: "Test User" },
      expires: "",
    };

    (useSession as jest.Mock).mockReturnValue({
      status: "authenticated",
      data: mockSession,
    });

    const { container, getByText } = render(<NavBar />);
    expect(container).toMatchSnapshot();
    expect(getByText('Test User')).toBeTruthy();
  });
});
