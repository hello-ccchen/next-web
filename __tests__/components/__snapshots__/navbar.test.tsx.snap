// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders \`Sign in\` button when user's session not exists 1`] = `
<div>
  <nav
    class="navbar fixed-top navbar-light bg-light shadow border-bottom border-light"
  >
    <div
      class="container-fluid"
    >
      <div
        class="d-flex justify-content-center align-items-center"
      >
        <a
          class="navbar-brand me-5"
          href="/"
        >
          <img
            alt="Home Logo"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="24"
            src="/next.svg"
            style="color: transparent;"
            width="100"
          />
        </a>
        <a
          href="/users"
        >
          User List
        </a>
      </div>
      <a
        href="/login"
      >
        Sign in
        <img
          alt="login"
          class="ms-2"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/login.svg"
          style="color: transparent;"
          width="24"
        />
      </a>
    </div>
  </nav>
</div>
`;

exports[`NavBar component renders logged-in user's info when user's session exists 1`] = `
<div>
  <nav
    class="navbar fixed-top navbar-light bg-light shadow border-bottom border-light"
  >
    <div
      class="container-fluid"
    >
      <div
        class="d-flex justify-content-center align-items-center"
      >
        <a
          class="navbar-brand me-5"
          href="/"
        >
          <img
            alt="Home Logo"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="24"
            src="/next.svg"
            style="color: transparent;"
            width="100"
          />
        </a>
        <a
          href="/users"
        >
          User List
        </a>
      </div>
      <div>
        Test User
        <img
          alt="login"
          class="ms-2"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/logout.svg"
          style="color: transparent; cursor: pointer;"
          width="24"
        />
      </div>
    </div>
  </nav>
</div>
`;
